Client/Frontend/Browser/OpenInHelper.swift

```diff
@@ -127,18 +132,24 @@ class OpenPdfInHelper: NSObject, OpenInHelper, UIDocumentInteractionControllerDe
     fileprivate var filepath: URL?
 
     required init?(response: URLResponse) {
-        guard let MIMEType = response.mimeType, MIMEType == MimeType.PDF.rawValue && UIApplication.shared.canOpenURL(URL(string: "itms-books:")!),
+        guard let MIMEType = response.mimeType,
+            (MIMEType == MimeType.PDF.rawValue || MIMEType == MimeType.EPUB.rawValue) && UIApplication.shared.canOpenURL(URL(string: "itms-books:")!),
             let responseURL = response.url else { return nil }
         url = responseURL
         super.init()
-        setFilePath(response.suggestedFilename ?? url.lastPathComponent )
+        setFilePath(response.suggestedFilename ?? url.lastPathComponent, mimeType: MIMEType)
     }
 
-    fileprivate func setFilePath(_ suggestedFilename: String) {
+    fileprivate func setFilePath(_ suggestedFilename: String, mimeType: String) {
         var filename = suggestedFilename
         let pathExtension = filename.asURL?.pathExtension
         if pathExtension == nil {
-            filename.append(".pdf")
+            if (mimeType == MimeType.PDF.rawValue) {
```

Comment:
I think we could write a method in `MimeType` that returns an extension based on the type. Something like 
`filename.append(MimeType(mimeType).extension)`. We could write a computed value in the enum like 
``` 
      var extension: String {
            switch self {
            case .PDF:
                 return “pdf”
            case .EPUP:
                 return "epub"
            }
       }
```